packages/ContentAutotagging/Engine/generic/AutotagBaseEngine.ts

```diff
@@ -0,0 +1,745 @@
+                const contentItemAttributes = <ContentItemAttributeEntity> await md.GetEntityObject('Content Item Attributes', contextUser)
+                
+                //Value should be a string, if its a null or undefined value, set it to an empty string
+                const value = LLMResults[key] || ''
+
+                contentItemAttributes.NewRecord()
+                contentItemAttributes.Set('ContentItemID', LLMResults.contentItemID)
+                contentItemAttributes.Set('Name', key)
+                contentItemAttributes.Set('Value', value)
+                await contentItemAttributes.Save()
+            }
+            // Overwrite name of content item with title if it exists
+            if (key === 'title') {
+                const ID = LLMResults.contentItemID
+                const contentItem = <ContentItemEntity> await md.GetEntityObject('Content Items', contextUser)
+                await contentItem.Load(ID)
+                contentItem.Set('Name', LLMResults.title)
+                await contentItem.Save()
+            }
+        }
+    }
+
+    /*** 
+    * Retrieves all of the content sources of a given content source type data from the database.
+    * @param contextUser: The user context to retrieve the content source data
+    * @returns A list of content sources
+    */
+    public async getAllContentSources(contextUser: UserInfo, contentSourceTypeID: string): Promise<ContentSourceEntity[]|undefined> {
+        const rv = new RunView();
+        
+        const contentSourceResult = await rv.RunView<ContentSourceEntity>({
+            EntityName: 'Content Sources',
+            ResultType: 'entity_object', 
+            ExtraFilter: `ContentSourceTypeID='${contentSourceTypeID}'`
+        }, contextUser);
+        try {
+        if (contentSourceResult.Success && contentSourceResult.Results.length) {
+            return contentSourceResult.Results
+        }
+        else {
+            throw new Error(`No content sources found for content source type with ID '${contentSourceTypeID}'`)
+        }
+        }
+    
+        catch (e) {
+            console.error(e);
+            throw e;
+            }
+    }
+
+    public async setSubclassContentSourceType(subclass: string, contextUser: UserInfo): Promise<string>{
+        const rv = new RunView()
+        const results = await rv.RunView<ContentSourceTypeEntity>({
+            EntityName: 'Content Source Types',
+            ExtraFilter: `Name='${subclass}'`,
+            ResultType: 'entity_object'
+        }, contextUser)
+        if (results.Success && results.Results.length) {
+            return results.Results[0].Get('ID')
+        }
+        else {
+            throw new Error(`Subclass with name ${subclass} not found`)
+        }
+    }
+
+    public async getLLMClient(modelID: string, contextUser: UserInfo): Promise<any> {
+        const modelDriverClass = await this.getModelDriverClassFromID(modelID, contextUser)
```

Comment:
We have this already built into AIEngine in a generic way, a hardcoded switch statement like the below should be a mental flag to say "there has to be a better way"

Use `export function GetAIAPIKey(AIDriverName: string): string` which is in the AIEngine package

@nico-oz  and @cadam11